packages/flutter_tools/lib/src/ios/migrations/ios_migrator.dart

```diff
@@ -54,3 +56,20 @@ class IOSMigrator {
     }
   }
 }
+
+class IOSMigration {
```

Comment:
Especially for a migration that might touch multiple files, this will have to be treated like a database transaction. We don't want to leave a tree in a partially migrated state, for example. Not for this PR (unless we already have multi-file migrations), but maybe add a TODO and file an issue.